Index: sundials-2.3.0/config/sundials-config.in
===================================================================
--- sundials-2.3.0.orig/config/sundials-config.in	2010-03-30 23:15:34.000000000 +0200
+++ sundials-2.3.0/config/sundials-config.in	2010-03-30 23:17:00.000000000 +0200
@@ -50,10 +50,10 @@
 shift $(( $OPTIND - 1 ))
 
 # args check
-[[ $module ]] || { echo >&2 the -m option and argument must be specified; exit 1; }
-[[ $vector ]] || { echo >&2 the -t option and argument must be specified; exit 1; }
-[[ $lang ]]   || { echo >&2 the -l option and argument must be specified; exit 1; }
-[[ $show ]]   || { show=both; }
+[ $module ] || { echo >&2 the -m option and argument must be specified; exit 1; }
+[ $vector ] || { echo >&2 the -t option and argument must be specified; exit 1; }
+[ $lang ]   || { echo >&2 the -l option and argument must be specified; exit 1; }
+[ $show ]   || { show=both; }
 
 # main
 
