Index: sundials-2.4.0/src/cvode/cvode.c
===================================================================
--- sundials-2.4.0.orig/src/cvode/cvode.c	2009-05-10 00:22:28.000000000 +0000
+++ sundials-2.4.0/src/cvode/cvode.c	2011-11-12 13:23:18.000000000 +0000
@@ -4128,7 +4128,7 @@
 
 #ifndef NO_FPRINTF_OUTPUT
     fprintf(stderr, "\n[%s ERROR]  %s\n  ", module, fname);
-    fprintf(stderr, msgfmt);
+    vfprintf(stderr, msgfmt, ap);
     fprintf(stderr, "\n\n");
 #endif
 
Index: sundials-2.4.0/src/cvodes/cvodes.c
===================================================================
--- sundials-2.4.0.orig/src/cvodes/cvodes.c	2009-05-10 00:22:35.000000000 +0000
+++ sundials-2.4.0/src/cvodes/cvodes.c	2011-11-12 13:26:19.000000000 +0000
@@ -8942,7 +8942,7 @@
 
 #ifndef NO_FPRINTF_OUTPUT
     fprintf(stderr, "\n[%s ERROR]  %s\n  ", module, fname);
-    fprintf(stderr, msg);
+    fprintf(stderr, "%s",msg);
     fprintf(stderr, "\n\n");
 #endif
 
Index: sundials-2.4.0/src/ida/ida.c
===================================================================
--- sundials-2.4.0.orig/src/ida/ida.c	2009-05-10 00:22:51.000000000 +0000
+++ sundials-2.4.0/src/ida/ida.c	2011-11-12 13:28:56.000000000 +0000
@@ -3248,7 +3248,7 @@
 
 #ifndef NO_FPRINTF_OUTPUT
     fprintf(stderr, "\n[%s ERROR]  %s\n  ", module, fname);
-    fprintf(stderr, msgfmt);
+    vfprintf(stderr, msgfmt,ap);
     fprintf(stderr, "\n\n");
 #endif
 
Index: sundials-2.4.0/src/idas/idas.c
===================================================================
--- sundials-2.4.0.orig/src/idas/idas.c	2009-05-10 00:22:58.000000000 +0000
+++ sundials-2.4.0/src/idas/idas.c	2011-11-12 13:30:16.000000000 +0000
@@ -7170,7 +7170,7 @@
 
 #ifndef NO_FPRINTF_OUTPUT
     fprintf(stderr, "\n[%s ERROR]  %s\n  ", module, fname);
-    fprintf(stderr, msgfmt);
+    vfprintf(stderr, msgfmt,ap);
     fprintf(stderr, "\n\n");
 #endif
 
Index: sundials-2.4.0/src/kinsol/kinsol.c
===================================================================
--- sundials-2.4.0.orig/src/kinsol/kinsol.c	2009-05-10 00:23:08.000000000 +0000
+++ sundials-2.4.0/src/kinsol/kinsol.c	2011-11-12 13:31:25.000000000 +0000
@@ -1864,7 +1864,7 @@
 
 #ifndef NO_FPRINTF_OUTPUT
     fprintf(stderr, "\n[%s ERROR]  %s\n  ", module, fname);
-    fprintf(stderr, msgfmt);
+    vfprintf(stderr, msgfmt,ap);
     fprintf(stderr, "\n\n");
 #endif
 
