Index: sundials-2.4.0/configure
===================================================================
--- sundials-2.4.0.orig/configure	2010-11-03 08:42:51.000000000 +0100
+++ sundials-2.4.0/configure	2010-11-03 08:43:14.000000000 +0100
@@ -8295,7 +8295,7 @@
 
           ;;
           # Ignore these flags.
-        -lang* | -lcrt*.o | -lc | -lgcc* | -lSystem | -libmil | -LANG:=* | -LIST:* | -LNO:*)
+        -lang* | -lcrt*.o | -lc | -lgcc* | -lSystem | -libmil | -little | -LANG:=* | -LIST:* | -LNO:*)
           ;;
         -lkernel32)
           test x"$CYGWIN" != xyes && ac_cv_f77_libs="$ac_cv_f77_libs $ac_arg"
