Index: sundials-2.5.0/bin/sundials-config.in
===================================================================
--- sundials-2.5.0.orig/bin/sundials-config.in	2012-04-12 02:31:04.000000000 +0200
+++ sundials-2.5.0/bin/sundials-config.in	2012-07-16 17:18:30.000000000 +0200
@@ -1,4 +1,4 @@
-#! @SHELL@
+#! /bin/bash
 # -----------------------------------------------------------------------------------
 
     NAME_="sundials-config"
@@ -65,7 +65,7 @@
 abs_includedir=`cd "${includedir}" > /dev/null 2>&1 && pwd`;
 abs_libdir=`cd "${libdir}" > /dev/null 2>&1 && pwd`;
 
-if test $abs_includedir != /usr/include ; then
+if test "$abs_includedir" != "/usr/include" ; then
     includes=-I$abs_includedir
 fi
 
Index: sundials-2.5.0/CMakeLists.txt
===================================================================
--- sundials-2.5.0.orig/CMakeLists.txt	2012-07-16 16:21:00.000000000 +0200
+++ sundials-2.5.0/CMakeLists.txt	2012-07-16 17:12:04.000000000 +0200
@@ -516,6 +516,10 @@
   ${PROJECT_SOURCE_DIR}/include/sundials/sundials_config.in
   ${PROJECT_BINARY_DIR}/include/sundials/sundials_config.h
   )
+CONFIGURE_FILE(
+  ${PROJECT_SOURCE_DIR}/bin/sundials-config.in
+  ${PROJECT_SOURCE_DIR}/bin/sundials-config
+  )
 
 # Add the include directory in the source tree and the one in
 # the binary tree (for the header file sundials_config.h)
